### PR TITLE
Add support for on-blur validation and normalization

### PR DESCRIPTION
**Pull request by Ameobea**
Due to problems caused by modifying form contents while typing, we migrate some functionality to take effect on blur.

### CODE REVIEW

File: src/components/apply/formNormalizers.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import _ from 'lodash';
+
+// pred is a function that accepts a value and returns true if it's valid and false otherwise.
```
Comment (jimmyhmiller) [inline src/components/apply/formNormalizers.js]: Comment not needed. `pred` is a pretty standard name.

File: src/components/apply/formNormalizers.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export const simpleNormalizer = pred => (input, prevInput) => pred(input) || input === '' ? input : prevInput;
+
+export const maxLengthValidator = len => input => input.length <= len;
+
+export const numericalValidator = input => !/[^0-9]/.test(input);
+
+export const stringValidator = input => !/[^A-z]/.test(input);
+
+// Returns a function that returns true if none of the validators reject the input.
+export const combineValidators = validators => input => _.isUndefined(_.find(validators, validator => !validator(input)));
```
Comment (jimmyhmiller) [inline src/components/apply/formNormalizers.js:12]: Are these validators going to be used for error messages?

File: src/reducers/formHelpers.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    normalize: (soc) => soc.substr(0,3) + '-' + soc.substr(3,2) + '-' + soc.substr(5,4),
+  },
+  [phoneNumber]: {
+    validate: input => /[0-9]+/.test(input) && input.length === 10,
+    normalize: normalizePhoneNumber,
+  },
+};
+
+export const blurFieldEntry = ( field, text ) => {
+  if( blurMap[field] && blurMap[field].validate(text) ){
```
Comment (jimmyhmiller) [inline src/reducers/formHelpers.js]: Not a big deal, but spacing here is weird.

**Review by jimmyhmiller — APPROVED** [review]

File: src/components/apply/formNormalizers.js (hunk 2 above)
Comment (claytn) [reply src/components/apply/formNormalizers.js:12]: No, the validators that will display error messages will be implemented at the form level and will be passed in directly through reduxForms.  The functions seen here are technically normalizers that will just limit users from typing invalid characters into specific field inputs as well as limiting the number of characters that can be entered.

File: src/reducers/formHelpers.js (hunk 3 above)
Comment (claytn) [reply src/reducers/formHelpers.js]: Fixed.

File: src/components/apply/formNormalizers.js (hunk 1 above)
Comment (claytn) [reply src/components/apply/formNormalizers.js]: Fixed.

**Pull request merged**